include/mm_types.h

```diff
@@ -237,6 +237,29 @@ typedef struct t_mmstream
 } mm_stream;
 //-----------------------------------------------------------------------------
 
+typedef struct t_mmsoundcnt_ds
+{
+    mm_word volume_mul : 7;
+    mm_word unused : 1;
+    mm_word volume_div : 2;
+    mm_word unused2 : 5;
+    mm_word hold : 1;
+    mm_word panning : 7;
+    mm_word unused3 : 1;
+    mm_word wave_duty : 3;
+    mm_word repeat_mode : 2;
+    mm_word format : 2;
+    mm_word status : 1;
+    mm_word src : 27;
+    mm_word unused4 : 5;
+    mm_hword timer;
+    mm_hword loop_start;
+    mm_word word_length : 22;
+    mm_word unused5 : 10;
+//-----------------------------------------------------------------------------
+} __attribute__((packed)) mmsoundcnt_ds;
```

Comment:
Btw, @AntonioND should we move this into libnds, since we were talking about that?

Reply:
libnds uses entirely different conventions for defining the same things, but *can* define them - I'm more in favor of the `#define` macros-to-registers style, personally, though I'm not sure if modern compilers have any performance differences with either approach.

Reply:
Yeah, it's more for having some type of structure instead of using things like (n << 4).

Reply:
Everything should have an appropriate libnds macro.

Reply:
If this is going to be unused going forward, it should be removed

Reply:
Commented away. Someone else may want to use it later on when dealing with mixing, so let's keep it there.

Reply:
I think I also prefer ``define`` macros, to be fair, specially when it comes to I/O registers. For regular data structures it would be fine.